control/veda_deliver_cielo.py

```diff
@@ -1,186 +1,228 @@
-            return None
-
-        c24_defaults = {
-            'c24_user': video_query.inst_class.c24_username,
-            'c24_pass': video_query.inst_class.c24_password,
-            'c24_speed': video_query.inst_class.c24_speed,
-            'c24_fidelity': video_query.inst_class.c24_fidelity,
-            'edx_id': self.veda_id,
+
+        cielo24_defaults = {
+            'api_key': self.video_transcript_preferences['api_key'],
+            'turnaround': self.video_transcript_preferences['cielo24_turnaround'],
+            'fidelity': self.video_transcript_preferences['cielo24_fidelity'],
+            'edx_id': self.video.edx_id,
             'url': url_query.encode_url
         }
-        return c24_defaults
-
-    def tokengenerator(self):
-        token_url = self.c24_site + self.c24_login + \
-            '?v=1&username=' + self.c24_defaults['c24_user'] + \
-            '&password=' + self.c24_defaults['c24_pass']
-
-        # Generate Token
-        r1 = requests.get(token_url)
-        if r1.status_code > 299:
-            ErrorObject.print_error(
-                message='Cielo24 API Access Error',
-            )
-            return None
-        api_token = ast.literal_eval(r1.text)["ApiToken"]
-        return api_token
-
-    def listjobs(self):
-        """List Jobs"""
-        api_token = self.tokengenerator()
-        r2 = requests.get(
-            ''.join((
-                self.c24_site,
-                self.c24_joblist,
-                '?v=1&api_token=',
-                api_token
-            ))
+        return cielo24_defaults
+
+    def start_transcription_flow(self):
+        """
+        Start cielo24 transcription flow.
+
+        This will do the following steps:
+        For each preferred language:
+            1. create a new job
+            2. add media url
+            3. perform transcript
+        """
+        job_id = None
+        preferred_langs = self.video_transcript_preferences.get('preferred_langs')
+
+        # update transcript status for video in edx-val
+        VALAPICall(video_proto=None, val_status=None).update_video_transcript_status(
+            self.video.studio_id, VideoTranscriptStatus.IN_PROGRESS, job_id
         )
-        job_list = r2.text
-        return job_list
 
-    def generate_jobs(self, api_token):
+        for perferred_lang in preferred_langs:
+            try:
+                job_id = self.create_job()
+                transcript_process_metadata = TranscriptProcessMetadata.objects.create(
+                    video=self.video,
+                    process_id=job_id,
+                    lang_code=perferred_lang,
+                    provider=TranscriptProvider.CIELO24,
+                    status=TranscriptStatus.IN_PROGRESS
+                )
+                self.create_add_media_url(job_id)
+                self.perform_transcript(job_id, perferred_lang)
+            except Cielo24Error:
+                transcript_process_metadata.status = TranscriptStatus.FAILED
```

Comment:
There might be a chance that `transcript_process_metadata` isn't declared yet – i.e. exception during `create_job`